gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagProcessingEngineMetrics.java

```diff
@@ -0,0 +1,155 @@
+  private ConcurrentMap<String, ContextAwareMeter>dagActionsRemovedFromStoreMeters =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter>dagActionsFailingRemovalMeters =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter>dagActionAverageProcessingDelayMillisMeters =  new ConcurrentHashMap();
+
+  public DagProcessingEngineMetrics(MetricContext metricContext) {
+    this.metricContext = metricContext;
+    registerAllMetrics();
+  }
+
+  public void registerAllMetrics() {
+    registerMetricForEachDagAction(this.dagActionsStoredMeters, ServiceMetricNames.DAG_ACTIONS_STORED);
+    registerMetricForEachDagAction(this.dagActionsObservedMeters, ServiceMetricNames.DAG_ACTIONS_OBSERVED);
+    registerMetricForEachDagAction(this.dagActionsLeasesObtainedMeters, ServiceMetricNames.DAG_ACTIONS_LEASES_OBTAINED);
+    registerMetricForEachDagAction(this.dagActionsNoLongerLeasingMeters, ServiceMetricNames.DAG_ACTIONS_NO_LONGER_LEASING);
+    registerMetricForEachDagAction(this.dagActionLeaseRemindersScheduledMeters, ServiceMetricNames.DAG_ACTION_LEASE_REMINDERS_SCHEDULED);
+    registerMetricForEachDagAction(this.dagActionRemindersProcessedMeters, ServiceMetricNames.DAG_ACTION_REMINDERS_PROCESSED);
+    registerMetricForEachDagAction(this.dagActionsExceededMaxRetryMeters, ServiceMetricNames.DAG_ACTIONS_EXCEEDED_MAX_RETRY);
+    registerMetricForEachDagAction(this.dagActionsInitFailedMeters, ServiceMetricNames.DAG_ACTIONS_INIT_FAILED);
+    registerMetricForEachDagAction(this.dagActionsInitSucceededMeters, ServiceMetricNames.DAG_ACTIONS_INIT_SUCCEEDED);
+    registerMetricForEachDagAction(this.dagActionExecutionsFailedMeters, ServiceMetricNames.DAG_ACTION_EXECUTIONS_FAILED);
+    registerMetricForEachDagAction(this.dagActionExecutionsSucceededMeters, ServiceMetricNames.DAG_ACTION_EXECUTIONS_SUCCEEDED);
+    registerMetricForEachDagAction(this.dagActionConclusionsFailedMeters, ServiceMetricNames.DAG_ACTION_CONCLUSIONS_FAILED);
+    registerMetricForEachDagAction(this.dagActionConclusionsSucceededMeters, ServiceMetricNames.DAG_ACTION_CONCLUSIONS_SUCCEEDED);
+  }
+
+  /**
+   * Create a meter of each dagActionType for the given metric, register it with the metric context, and store it in a
+   * concurrent map.
+   * @param metricMap
+   * @param metricName
+   */
+  private void registerMetricForEachDagAction(ConcurrentMap<String, ContextAwareMeter> metricMap, String metricName) {
+    for (DagActionStore.DagActionType dagActionType : DagActionStore.DagActionType.values()) {
+      metricMap.put(dagActionType.toString(),
+          this.metricContext.contextAwareMeter(metricName + dagActionType));
+    }
+  }
+
+  /**
+   * Updates the meter corresponding to the metricsName and dagActionType provided if they match an existing metric
+   * @param metricName
+   * @param dagActionType
+   */
+  public void updateMetricForDagAction(String metricName, DagActionStore.DagActionType dagActionType) {
```

Comment:
a set of type-safe methods would be so preferable - a lot can go wrong w/ passing in `String metricName`!

e.g. instead of having a massive switch statement buried within the impl which also pushes complexity (and the potential for errors) to callers, by making them hard-code a reference to a well-known constant at each point of use:
```
this.dagProcEngineMetrics.updateMetricForDagAction(ServiceMetricNames.DAG_ACTION_REMINDERS_PROCESSED,
                    leaseParams.getDagAction());
```
instead ask them to hard-code a type-safe and semantic method name:
```
this.dagProcEngineMetrics.markReminderProcessed(leaseParams.getDagAction());
```
or
```
this.dagProcEngineMetrics.incrementInitializeSucceeded(dagAction);
```

the impl should be trivial:
```
public void markReminderProcessed(DagAction dagAction) {
  updateMetric(this.reminderProcessedByDagActionType, dagAction.getDagActionType());
}
```